cocos/renderer/CCMaterial.cpp

```diff
@@ -339,7 +345,22 @@ bool Material::parseShader(Pass* pass, Properties* shaderProperties)
 
         auto vertShaderSrc = fu->getStringFromFile(vertShader);
         auto fragShaderSrc = fu->getStringFromFile(fragShader);
-        std::string defs = compileTimeDefines;
+
+        auto defineParts = Console::Utility::split(compileTimeDefines, ';');
+        std::stringstream ss;
+        for (auto &p : defineParts)
+        {
+            if (p.find("#define ") == std::string::npos)
+            {
+                ss << "#define " << p << std::endl;
+            }
+            else
+            {
+                ss << p << std::endl;
+            }
+        }
+
+        std::string defs = ss.str();
```

Comment:
Why need these codes?

Reply:
It was originally done here
https://github.com/cocos2d/cocos2d-x/blob/f0f732cc90c0ac18cbe2e4644f69c2bbfcc6a2bd/cocos/renderer/CCGLProgram.cpp#L43-L72

Reply:
Got it, but put it into a function make the codes more readable.

Reply:
okay